### PR TITLE
Transfer rabbit mq from app to inf layer and fix add components

### PR DESCRIPTION
**Pull request by vladzhorov**

### CODE REVIEW

File: ServiceFinder.OrderService/ServiceFinder.OrderService.Infrastructure/InfrasctructureExtension.cs [hunk 1]
```diff
@@ -11,17 +17,51 @@ public static class InfrastructureExtension
...
+                return channel;
+            });
+
+            services.AddScoped<IMessagePublisher, MessagePublisher>();
             services.AddScoped<IOrderRepository, OrderRepository>();
             services.AddScoped<IOrderRequestRepository, OrderRequestRepository>();
+            services.AddScoped<IDateTimeProvider, DateTimeProvider>();
+            services.AddScoped<IDomainEventDispatcher, DomainEventDispatcher>();
+            services.AddScoped<Domain.Services.OrderService>();
+            services.AddScoped<OrderRequestService>();
```
Comment (moltenessence) [inline ServiceFinder.OrderService/ServiceFinder.OrderService.Infrastructure/InfrasctructureExtension.cs:63]: where is interface?

File: ServiceFinder.OrderService/ServiceFinder.OrderService.Infrastructure/InfrasctructureExtension.cs [hunk 2]
```diff
@@ -11,17 +17,51 @@ public static class InfrastructureExtension
...
+
+                return channel;
+            });
+
+            services.AddScoped<IMessagePublisher, MessagePublisher>();
             services.AddScoped<IOrderRepository, OrderRepository>();
             services.AddScoped<IOrderRequestRepository, OrderRequestRepository>();
+            services.AddScoped<IDateTimeProvider, DateTimeProvider>();
+            services.AddScoped<IDomainEventDispatcher, DomainEventDispatcher>();
+            services.AddScoped<Domain.Services.OrderService>();
```
Comment (moltenessence) [inline ServiceFinder.OrderService/ServiceFinder.OrderService.Infrastructure/InfrasctructureExtension.cs:62]: why domain service here?

**Review by moltenessence — CHANGES_REQUESTED** [review]

**Review by Mmishaaa — APPROVED** [review]

File: ServiceFinder.OrderService/ServiceFinder.OrderService.Application/Interfaces/IOrderAppService.cs [hunk 3]
```diff
@@ -8,5 +8,6 @@ public interface IOrderAppService
         Task<OrderDto> CreateOrderAsync(OrderDto orderDTO, decimal baseRatePerMinute, int baseRateDurationInMinutes, CancellationToken cancellationToken);
         Task UpdateOrderStatusAsync(Guid orderId, OrderStatus newStatus, CancellationToken cancellationToken);
         Task<OrderDto> GetOrderByIdAsync(Guid id, CancellationToken cancellationToken);
+        Task<PagedResult<OrderDto>> GetAllOrderAsync(int pageNumber, int pageSize);
```
Comment (moltenessence) [inline ServiceFinder.OrderService/ServiceFinder.OrderService.Application/Interfaces/IOrderAppService.cs:11]: no cancellation token?

File: ServiceFinder.OrderService/ServiceFinder.OrderService.Application/ApplicationExtension.cs [hunk 4]
```diff
@@ -1,54 +1,22 @@
...
             services.AddScoped<IOrderRequestStatusChangedEventHandler, OrderRequestStatusChangedEventHandler>();
             services.AddScoped<IOrderStatusChangedEventHandler, OrderStatusChangedEventHandler>();
-        }
-
-        public static void AddApplication(this IServiceCollection services)
-        {
             services.AddScoped<IOrderAppService, OrderAppService>();
             services.AddScoped<IOrderRequestAppService, OrderRequestAppService>();
+            services.AddScoped<IDomainEventDispatcher, DomainEventDispatcher>();
+            services.AddScoped<IOrderService, Domain.Services.OrderService>();
```
Comment (moltenessence) [inline ServiceFinder.OrderService/ServiceFinder.OrderService.Application/ApplicationExtension.cs:18]: why domain service here?

File: ServiceFinder.OrderService/ServiceFinder.OrderService.Application/Interfaces/IOrderRequestAppService.cs [hunk 5]
```diff
@@ -8,5 +9,6 @@ public interface IOrderRequestAppService
         Task<OrderRequestDto> CreateOrderRequestAsync(OrderRequestDto orderRequestDTO, CancellationToken cancellationToken);
         Task UpdateOrderRequestStatusAsync(Guid orderRequestId, OrderRequestStatus newStatus, CancellationToken cancellationToken);
         Task<OrderRequestDto> GetOrderRequestByIdAsync(Guid id, CancellationToken cancellationToken);
+        Task<PagedResult<OrderRequestDto>> GetAllOrderRequestAsync(int pageNumber, int pageSize);
```
Comment (moltenessence) [inline ServiceFinder.OrderService/ServiceFinder.OrderService.Application/Interfaces/IOrderRequestAppService.cs:12]: cancellation token

**Review by moltenessence — CHANGES_REQUESTED** [review]

**Review by moltenessence — APPROVED** [review]

**Pull request merged**